src/silly_sprites/event/base_handler.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "event_concept.hpp"
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+
+namespace sly::event {
+    template<Event E>
+    class Handler {
+    public:
+        Handler() {
+            E::connect(*this);
+        };
+        Handler(Handler const& other) = delete;
+        Handler(Handler&& other) noexcept = delete;
+        Handler& operator=(Handler const& other) = delete;
+        Handler& operator=(Handler&& other) = delete;
+
+        virtual ~Handler() {
+            E::disconnect(*this);
+        }
+
+        virtual void on_event(E const& event) = 0;
+    };
+
+    template<Event E>
+    struct Base {
+    private:
+        inline static std::vector<Handler<E>*> s_handlers;
```

Comment:
I'm wondering if an `std::unordered_set` would be more reasonable here since it automatically doesn't allow duplicates. Then you wouldn't need `contains_handler` anymore (I think).